http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dns/ProxyResolver.scala

```diff
@@ -52,7 +52,7 @@ object ProxyResolver {
         mapping <- hostToIpMapping.get
         // Use the IP if we have a mapping for it; otherwise fall back to default host name resolution
         h = mapping.get(host).map(_.asString).getOrElse(host)
-        res <- F.delay(new InetSocketAddress(h, port))
+        res <- F.blocking(new InetSocketAddress(h, port))
```

Comment:
I think this was kind of pathologically bad in Back Leo. Here's the thread dump:
```
[WARNING] A Cats Effect worker thread was detected to be in a blocked state (BLOCKED)
at java.base@17.0.8.1/java.net.InetAddress$NameServiceAddresses.get(Unknown Source)
at java.base@17.0.8.1/java.net.InetAddress.getAllByName0(Unknown Source)
at java.base@17.0.8.1/java.net.InetAddress.getAllByName(Unknown Source)
at java.base@17.0.8.1/java.net.InetAddress.getAllByName(Unknown Source)
at java.base@17.0.8.1/java.net.InetAddress.getByName(Unknown Source)
at java.base@17.0.8.1/java.net.InetSocketAddress.<init>(Unknown Source)
at app//org.broadinstitute.dsde.workbench.leonardo.dns.ProxyResolver$ProxyResolverInterp.$anonfun$resolveInternal$4(ProxyResolver.scala:55)
at app//org.broadinstitute.dsde.workbench.leonardo.dns.ProxyResolver$ProxyResolverInterp$$Lambda$3984/0x00007fcb490022f0.apply(Unknown Source)
at app//cats.effect.IOFiber.runLoop(IOFiber.scala:255)
at app//cats.effect.IOFiber.execR(IOFiber.scala:1360)
at app//cats.effect.IOFiber.run(IOFiber.scala:112)
at app//cats.effect.unsafe.WorkerThread.run(WorkerThread.scala:632)
```

`InetSocketAddress` is a network call, and was being triggered a LOT but the metrics collection process (which http-pings every runtime/app). I have a hunch this was contributing to Back Leo instability.

Reply:
didn't realize this tries to resolve the actual url 😵‍💫